Core/Core/Inbox/ViewModel/InboxViewModel.swift

```diff
@@ -135,6 +143,19 @@ public class InboxViewModel: ObservableObject {
             .map { interactor.updateState(message: $0.message, state: $0.state) }
             .sink()
             .store(in: &subscriptions)
+
+        starDidTap
+            .map { [messageInteractor] starred, messageId in
+                messageInteractor.updateStarred(starred: starred, messageId: messageId) }
+            .sink()
```

Comment:
I may be mistaken, but couldn't the code inside the map go directly to the sink instead?
Same for next chain below

Reply:
I will use flatMap cuz the  messageInteractor.updateStarred return a publisher